xbmc/filesystem/File.cpp

```diff
@@ -283,7 +288,10 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
         return m_pFile->Open(url);
       }
     }
-    m_pFile = CFileFactory::CreateLoader(url);
+    if (m_pFile)
+      m_pFile->Close();
+    else
+      m_pFile = CFileFactory::CreateLoader(url);
```

Comment:
Ideally yes, but still I doubt everyone will understand why this is, the way it is now. That's why I'd like a comment.